### ISSUE TITLE
How to define custom event modifiers in Aurelia 2?

### ISSUE THREAD

**Issue by HamedFathi** [aurelia/aurelia#796]
# 🙋 Feature Request

As recently mentioned in a [blog post](https://aurelia.io/blog/2019/10/31/aurelia-vnext-2019-fall-update), Aurelia 2 has wonderful extensibility capability. We can even simulate the syntax of other frameworks. This is really amazing.

Vue and Angular introduced a feature as `Event binding`. You can find some resources as the following links:

[Vue Event Modifiers](https://vuejs.org/v2/guide/events.html#Event-Modifiers)

```html
<form v-on:submit.prevent="onSubmit"></form>
```

[Vue Key Modifiers](https://vuejs.org/v2/guide/events.html#Key-Modifiers)

```html
<input v-on:keyup.page-down="onPageDown">
```

[Vue System Modifier Keys](https://vuejs.org/v2/guide/events.html#System-Modifier-Keys)

```html
<!-- Alt + C -->
<input @keyup.alt.67="clear">

<!-- Ctrl + Click -->
<div @click.ctrl="doSomething">Do something</div>
```

Even `Vue` lets you declare [custom key modifier](https://vuejs.org/v2/api/#keyCodes)

```ts
Vue.config.keyCodes = {
  v: 86,
  f1: 112,
  // camelCase won`t work
  mediaPlayPause: 179,
  // instead you can use kebab-case with double quotation marks
  "media-play-pause": 179,
  up: [38, 87]
}

<input type="text" @keyup.media-play-pause="method">
```

[Event Modifiers for Angular Applications](https://github.com/NetanelBasal/ngx-event-modifiers)

```ts
import { EventModifiersModule } from 'ngx-event-modifiers';

@NgModule({
  imports: [
    EventModifiersModule
  ]
})
export class AppModule { }

<button (click.stop)="onClick($event, extraData)">Click Me!!</button>
<button (click.prevent)="onClick($event, extraData)" type="submit">Click Me!!</button>
```

Is it possible to have/implement this capability in Aurelia 2? As a plugin or even a native feature?

**Comment by fkleuver** [comment]
Yep, this can be implemented with our `@bindingCommand` extensibility point. I'm marking this issue as a reminder that I need to add some examples to the docs for this. We could decide later on if it makes sense to add it to the core.

**Comment by EisenbergEffect** [comment]
For reference, just add a child page here: https://docs.aurelia.io/examples/binding-and-templating with the code and brief explanation. I think it's valuable to keep there even if we do add it to the framework as it's a great way for people to learn how these things can be built and plugged in.
